lib/sanitize.rb

```diff
@@ -229,14 +231,18 @@ def transform_node!(node, node_whitelist)
       config = @transformer_config
       config[:is_whitelisted] = node_whitelist.include?(node)
       config[:node] = node
-      config[:node_name] = node.name.downcase
+      config[:node_name] = node_name
       config[:node_whitelist] = node_whitelist
 
       result = transformer.call(config)
 
       if result.is_a?(Hash) && result[:node_whitelist].respond_to?(:each)
         node_whitelist.merge(result[:node_whitelist])
       end
+
+      # It's possible that a transformer modifies the node name, so only
+      # copy the string again if it has changed.
+      node_name = node.name.downcase if node.name != node_name
```

Comment:
We need to account for the possibility that `node.name` and `node_name` won't have the same case.

One possible optimization would be to compare the downcased node name with the original name at the start of `transform_node!` (and whenever the node name changes) and use that knowledge to decide whether we need to do a case-insensitive equality check in future iterations of the loop (probably using `casecmp`).

My guess is that 99+% of the time this would let us skip using `casecmp` in the loop.